### PR TITLE
IVS-205 - Limit number of outcomes per rule

### PR DESCRIPTION
**Pull request by rw-bsi**
Limit # of outcomes to 10 (configurable) + moved some of grouping/title logic to backend

Example:
image

### CODE REVIEW

**Review by evandroAlfieri — APPROVED** [review]

File: backend/apps/ifc_validation_bff/views_legacy.py [hunk 1]
```diff
@@ -435,6 +438,15 @@ def report(request, id: str):
                     "msg": outcome.observed,
                     "task_id": outcome.validation_task_public_id
                 }
+
+                key = mapped['attribute'] if 'attribute' in mapped and mapped['attribute'] else 'Uncategorized'
```
Comment (aothms) [inline backend/apps/ifc_validation_bff/views_legacy.py:442]: I had read this several times, can this be abbreviated to:

```suggestion
                key = mapped.get('attribute', None) or 'Uncategorized'
```

File: backend/apps/ifc_validation_bff/views_legacy.py [hunk 2]
```diff
@@ -418,14 +419,16 @@ def report(request, id: str):
         logger.info('Fetching and mapping syntax done.')
 
     # retrieve and map schema outcome(s) + instances
+    schema_results_count = {}
```
Comment (aothms) [inline backend/apps/ifc_validation_bff/views_legacy.py:422]:
```suggestion
    schema_results_count = defaultdict(int)
```

File: backend/apps/ifc_validation_bff/views_legacy.py [hunk 3]
```diff
@@ -435,6 +438,15 @@ def report(request, id: str):
                     "msg": outcome.observed,
                     "task_id": outcome.validation_task_public_id
                 }
+
+                key = mapped['attribute'] if 'attribute' in mapped and mapped['attribute'] else 'Uncategorized'
+                _type = mapped['constraint_type'] if 'constraint_type' in mapped and mapped['constraint_type'] else 'Uncategorized'
+                title = _type.replace('_', ' ').capitalize() + ' - ' + key
+                mapped['title'] = title # eg. 'Schema - SegmentStart'
+                schema_results_count[title] = schema_results_count[title] + 1 if title in schema_results_count else 1
```
Comment (aothms) [inline backend/apps/ifc_validation_bff/views_legacy.py:446]:
```suggestion
                schema_results_count[title] += 1
```

File: backend/apps/ifc_validation_bff/views_legacy.py [hunk 4]
```diff
@@ -453,14 +465,19 @@ def report(request, id: str):
         ('prerequisites', (ValidationTask.Type.PREREQUISITES,)),
         ('industry', (ValidationTask.Type.INDUSTRY_PRACTICES,)))
     
+    grouped_gherkin_outcomes_counts = { 
+        'normative': {},
+        'prerequisites': {},
+        'industry': {}
```
Comment (aothms) [inline backend/apps/ifc_validation_bff/views_legacy.py:471]:
```suggestion
        'normative': defaultdict(int),
        'prerequisites': defaultdict(int),
        'industry': defaultdict(int),
```

File: backend/apps/ifc_validation_bff/views_legacy.py [hunk 5]
```diff
@@ -482,6 +499,14 @@ def report(request, id: str):
                 "task_id": outcome.validation_task_public_id,
                 "msg": outcome.observed,                    
             }
+            
+            # TODO: organize this differently?
+            key = 'Schema - Version' if label == 'prerequisites' else mapped['feature']
+            mapped['title'] = key
+            grouped_gherkin_outcomes_counts[label][key] = grouped_gherkin_outcomes_counts[label][key] + 1 if key in grouped_gherkin_outcomes_counts[label] else 1
```
Comment (aothms) [inline backend/apps/ifc_validation_bff/views_legacy.py:506]:
```suggestion
            grouped_gherkin_outcomes_counts[label][key] = += 1
```

**Review by aothms — APPROVED** [review]
Nicely done! Courageous to touch this messy logic and move it around. Only some cosmetic things from my side.

File: backend/apps/ifc_validation_bff/views_legacy.py (hunk 5 above)
Comment (rw-bsi) [reply backend/apps/ifc_validation_bff/views_legacy.py]: Minor typo but replaced with `+= 1`

**Pull request merged**